ui/vic-ui-h5c/vic-service/src/main/java/com/vmware/vic/utils/VicVmComparator.java

```diff
@@ -51,20 +54,69 @@ public VicVmComparator(
 	 */
 	@Override
 	public int compare(String a, String b) {
-		try {
-		    ModelObject mo_A = (ModelObject) base.get(a);
-	        ModelObject mo_B = (ModelObject) base.get(b);
-			int result = 0;
-			result = getStringPropertyFromVm(mo_A)
-				.compareTo(getStringPropertyFromVm(mo_B));
-			if (result == 0) {
-			    result = mo_A.hashCode() - mo_B.hashCode();
-			}
+	    ModelObject mo_A = (ModelObject) base.get(a);
+        ModelObject mo_B = (ModelObject) base.get(b);
 
-			return result * (this.reverse ? -1 : 1);
-		} catch (IndexOutOfBoundsException e) {
-			return 0;
-		}
+        return compareValues(mo_A, mo_B);
+	}
+
+	/**
+	 * Compare ModelObjects a and b based on compareBy and reverse.
+	 * This is basically a compareTo() & compare() wrapper for string,
+     * integer, long, and double types 
+	 * @param mo_A : ModelObject a
+	 * @param mo_B : ModelObject b
+	 * @return compare result
+	 */
+	private int compareValues(ModelObject mo_A, ModelObject mo_B) {
+	    int result = 0;
+	    String valueA = getStringPropertyFromVm(mo_A);
+        String valueB = getStringPropertyFromVm(mo_B);
+	    Number numA = tryParseNumericValue(valueA);
+	    Number numB = tryParseNumericValue(valueB);
+
+	    // if either numA or numB ends up null then string comparison is used
+	    if (numA == null || numB == null) {
```

Comment:
I had to sneak in some small changes in Java files with this PR. Sorry for the confusion